federation/ldap/src/main/java/org/keycloak/services/managers/LDAPServerCapabilitiesManager.java

```diff
@@ -94,6 +95,29 @@ public static LDAPConfig buildLDAPConfig(TestLdapConnectionRepresentation config
         return new LDAPConfig(configMap);
     }
 
+    /**
+     * Ensure provided connection URI matches parsed LDAP connection URI.
+     *
+     * See: https://docs.oracle.com/javase/jndi/tutorial/ldap/misc/url.html
+     * @param config
+     * @param ldapConfig
+     * @return
+     * @throws ModelValidationException if an invalid URL is provided
+     */
+    private static boolean checkLdapConnectionUrl(TestLdapConnectionRepresentation config, LDAPConfig ldapConfig) {
+        // There could be multiple connection URIs separated via spaces.
+        String[] configConnectionUrls = config.getConnectionUrl().split(" ");
+        String[] ldapConfigConnectionUrls = ldapConfig.getConnectionUrl().split(" ");
+        if (configConnectionUrls.length != ldapConfigConnectionUrls.length) {
+            throw new ModelValidationException("invalid ldap url");
```

Comment:
Perhaps a little more information can be provided here as to why this URL is invalid?